PlutoHelperAgent/AppDelegate.m

```diff
@@ -125,92 +125,150 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
+                                    });
+                                }
+                            }];
+                            [task launch];
                         }
-                    }];
-                    [task launch];
-                }
-            } errorHandler:errorHandlerBlock];
-        } errorHandler:errorHandlerBlock];
+                    } errorHandler:errorHandlerBlock];
+                } errorHandler:errorHandlerBlock];
+            } else if (connectionStatus == PERMISSION_DENIED) {
+                NSAlert *alert = [[NSAlert alloc] init];
+                
+                [alert addButtonWithTitle:@"Okay"];
+                
+                NSString *message = [NSString stringWithFormat: @"Permission Denied"];
+                
+                [alert setMessageText:message];
+                
+                [alert setInformativeText:@"Please check you have entered your username and password correctly."];
+                
+                [alert setAlertStyle:NSWarningAlertStyle];
+                
+                [NSApp setActivationPolicy:NSApplicationActivationPolicyAccessory];
+                [[self prefsWindow] setLevel:NSFloatingWindowLevel];
+                [[[self prefsWindow] windowController] showWindow:self];
+                
+                if ([alert runModal] == NSAlertFirstButtonReturn) {
 
+                }
+            } else if (connectionStatus == SERVER_ERROR) {
+                NSAlert *alert = [[NSAlert alloc] init];
+                
+                [alert addButtonWithTitle:@"Okay"];
+                
+                NSString *message = [NSString stringWithFormat: @"Server Error"];
+                
+                [alert setMessageText:message];
+                
+                [alert setInformativeText:@"An error occurred on the server. Please report this error to multimediatech@theguardian.com"];
+                
+                [alert setAlertStyle:NSWarningAlertStyle];
+                
+                if ([alert runModal] == NSAlertFirstButtonReturn) {
+                    
+                }
+            } else {
+                NSAlert *alert = [[NSAlert alloc] init];
+                
+                [alert addButtonWithTitle:@"Okay"];
+                
+                NSString *message = [NSString stringWithFormat: @"Unknown Error"];
+                
+                [alert setMessageText:message];
+                
+                [alert setInformativeText:@"An unknown error occurred. Please report this error to multimediatech@theguardian.com"];
+                
+                [alert setAlertStyle:NSWarningAlertStyle];
+                
+                if ([alert runModal] == NSAlertFirstButtonReturn) {
+                    
+                }
+            }
+        } errorHandler:^(NSURLResponse *response, NSError *error) {
+            NSLog(@"An error occurred when attempting to check if the user is logged in.");
```

Comment:
Please show an error to the user indicating what went wrong, and dump detailed error information to the log